pkg/security/resolvers/netns/resolver.go

```diff
@@ -156,11 +156,12 @@ func (nn *NetworkNamespace) dequeueNetworkDevices(tcResolver *tc.Resolver, manag
 	if err != nil {
 		return
 	}
-	defer handle.Close()
-
 	for _, queuedDevice := range nn.networkDevicesQueue {
 		_ = tcResolver.SetupNewTCClassifierWithNetNSHandle(queuedDevice, handle, manager)
 	}
+	if err := handle.Close(); err != nil {
```

Comment:
since we are only logging here (and not playing with return value) this could go in a defer function. `defer` also brings semantic value of "this should be closed in all code paths after this point", removing it could allow a future code change to miss the close (by returning early on an error case as an example, or by panicking)